libs/map-editor/src/GameMap/GameMapEntities.ts

```diff
@@ -0,0 +1,133 @@
+        if (!entity) {
+            throw new Error(`Entity of id: ${id} does not exists!`);
+        }
+        _.merge(entity, config);
+        this.updateEntityInMapProperties(entity);
+        return entity;
+    }
+
+    private addEntityToMapProperties(entityData: EntityData): boolean {
+        if (this.gameMap.getMap().properties === undefined) {
+            this.gameMap.getMap().properties = [];
+        }
+        if (!this.getEntitiesMapProperty()) {
+            this.gameMap.getMap().properties?.push({
+                name: this.MAP_PROPERTY_ENTITIES_NAME,
+                type: "class",
+                value: JSON.parse(JSON.stringify([])) as Json,
+            });
+        }
+        const entitiesPropertyValues = JSON.parse(JSON.stringify(this.getEntitiesMapProperty()?.value)) as EntityData[];
+
+        if (entitiesPropertyValues.find((entity) => entity.id === entityData.id)) {
+            console.warn(`ADD ENTITY FAIL: ENTITY OF ID ${entityData.id} ALREADY EXISTS WITHIN THE GAMEMAP!`);
+            return false;
+        }
+        entitiesPropertyValues.push(entityData);
+
+        const entitiesMapProperty = this.getEntitiesMapProperty();
+        if (entitiesMapProperty !== undefined) {
+            entitiesMapProperty.value = JSON.parse(JSON.stringify(entitiesPropertyValues)) as Json;
+        }
+
+        return true;
+    }
+
+    private deleteEntityFromMapProperties(id: number): boolean {
+        const entitiesPropertyValues = JSON.parse(JSON.stringify(this.getEntitiesMapProperty()?.value)) as EntityData[];
+        const indexToRemove = entitiesPropertyValues.findIndex((entityData) => entityData.id === id);
+        if (indexToRemove !== -1) {
+            entitiesPropertyValues.splice(indexToRemove, 1);
+            const entitiesMapProperty = this.getEntitiesMapProperty();
+            if (entitiesMapProperty !== undefined) {
+                entitiesMapProperty.value = JSON.parse(JSON.stringify(entitiesPropertyValues)) as Json;
+                return true;
+            } else {
+                return false;
+            }
+        }
+        return false;
+    }
+
+    private updateEntityInMapProperties(entityData: EntityData): boolean {
+        const entitiesPropertyValue = JSON.parse(JSON.stringify(this.getEntitiesMapProperty()?.value)) as EntityData[];
+
+        const entityIndex = entitiesPropertyValue.findIndex((entity) => entity.id === entityData.id);
+
+        if (entityIndex === -1) {
+            console.warn(`CANNOT FIND ENTITY WITH ID: ${entityData.id} IN MAP PROPERTIES!`);
+            return false;
+        }
+
+        const entitiesMapProperty = this.getEntitiesMapProperty();
+        if (entitiesMapProperty !== undefined) {
+            entitiesPropertyValue[entityIndex] = entityData;
+            entitiesMapProperty.value = JSON.parse(JSON.stringify(entitiesPropertyValue)) as Json;
+            return true;
+        }
+        return false;
+    }
```

Comment:
Making entities property only update when saving the map currently creates a case where player manipulates entities somehow, reloads the page and loses these changes because all he gets when asking for the map is an ITiledMap JSON.

I need to thinkg about a possible solution to this.